### PR TITLE
New run matlab command

### PR DESCRIPTION
**Pull request by davidbuzinski**
Update to use the new binary for run-matlab-command.

### CODE REVIEW

File: src/matlab.ts [hunk 1]
```diff
@@ -67,7 +67,22 @@ export async function runCommand(hs: HelperScript, platform: string, fn: ExecFn)
  * @param platform Operating system of the runner (e.g., "win32" or "linux")
  */
 export function getRunMATLABCommandScriptPath(platform: string): string {
-    const ext = platform === "win32" ? "bat" : "sh";
-    const rmcPath = path.join(__dirname, "bin", `run_matlab_command.${ext}`);
+    let ext;
+    let platform_dir;
```
Comment (mcafaro) [inline src/matlab.ts:71]: The rest of this code looks like it's using camelCase.

File: scripts/setupdeps.sh [hunk 2]
```diff
@@ -1,17 +1,29 @@
...
+for os in ${SUPPORTED_OS[@]}
+do
+    if [[ $os == 'win64' ]] ; then
+        bin_ext='.exe'
+    else
+        bin_ext=''
+    fi
+    mkdir -p "$WORKINGDIR/$os"
+    wget -O  "$WORKINGDIR/$os/run-matlab-command$bin_ext" "$RMC_BASE_URL/$os/run-matlab-command$bin_ext"
+    mv -f ./* "$DISTDIR/"
```
Comment (mcafaro) [inline scripts/setupdeps.sh:26]: Can this just be done once after the loop? Or should the specific folder we want to move be specified here?

File: src/matlab.ts [hunk 3]
```diff
@@ -67,7 +67,22 @@ export async function runCommand(hs: HelperScript, platform: string, fn: ExecFn)
...
+            ext = ".exe";
+            platform_dir = "win64";
+            break;
+        case "darwin":
+            ext = "";
+            platform_dir = "maci64";
+            break;
+        default:
+            ext = "";
+            platform_dir = "glnxa64";
```
Comment (mcafaro) [inline src/matlab.ts:83]: There's a bit of a disconnect here between platform and architecture. Maybe this function should take in the architecture as well so we can be sure we're picking the right binary or erroring if we don't have a binary for that combination.

Comment (davidbuzinski) [reply src/matlab.ts:83]: Hmmm yeah that's a good point. Is there any architectures beyond x64 that we know to be in use currently? Could this technically be a "breaking change" for some self-hosted runners?

File: src/matlab.ts [hunk 4]
```diff
@@ -65,9 +66,31 @@ export async function runCommand(hs: HelperScript, platform: string, fn: ExecFn)
...
  * @param platform Operating system of the runner (e.g., "win32" or "linux")
- */
-export function getRunMATLABCommandScriptPath(platform: string): string {
-    const ext = platform === "win32" ? "bat" : "sh";
-    const rmcPath = path.join(__dirname, "bin", `run_matlab_command.${ext}`);
+ * @param architecture Architecture of the runner (e.g., "x64")
+*/
+export function getRunMATLABCommandScriptPath(platform: string, architecture: string): string {
+    if (architecture != "x64") {
+        throw new Error(`This action is not supported for runners of platform ${platform} and architecture ${architecture}.`);
```
Comment (davidbuzinski) [inline src/matlab.ts:73]: @mw-hrastega any thoughts on this error message?

Comment (mw-hrastega) [reply src/matlab.ts:73]: Could you give me an example of the message holes? Will each hole be filled with a scalar or a list?

If both are scalar, maybe something like this?

This action is not supported on ${platform} runners using the ${architecture} architecture.

Comment (davidbuzinski) [reply src/matlab.ts:73]: I think that sounds reasonable. Yes, they will both be scalar. One example of such an error would read `This action is not supported on windows runners using the x86 architecture.`

Comment (mw-hrastega) [reply src/matlab.ts]: Sounds good to me.

**Review by mw-hrastega — APPROVED** [review]

File: src/matlab.ts (hunk 3 above)
Comment (mcafaro) [reply src/matlab.ts]: It looks like GitHub only [supports x64 runners (and ARM32/64 on Linux)](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners#supported-architectures-and-operating-systems-for-self-hosted-runners), so we should be safe providing just x64 binaries. I think just from a code clarity and correctness standpoint, we should still take the arch into consideration here and just error if it doesn't match our expectations.

**Review by mcafaro — APPROVED** [review]

**Pull request merged**